pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/JsonExtractIndexTransformFunction.java

```diff
@@ -44,7 +44,7 @@ public class JsonExtractIndexTransformFunction extends BaseTransformFunction {
   private TransformResultMetadata _resultMetadata;
   private JsonIndexReader _jsonIndexReader;
   private Object _defaultValue;
-  private Map<String, RoaringBitmap> _matchingDocsMap;
+  private Map<String, RoaringBitmap> _valueToMatchingFlattenedDocIdsMap;
```

Comment:
I just realized that we should keep both valueToDocId and valueToFlattenDocId. Using valueToFlattenDocIds would increase the read latency since it need to iterate the flattenDocId bitmap and create the docId bitmap. If the return type isSingleValue, init valueToDocId; otherwise, init valueToFlattenDocIds.